cluster-autoscaler/provisioningrequest/orchestrator/orchestrator_test.go

```diff
@@ -263,14 +415,26 @@ func TestScaleUp(t *testing.T) {
 					t.Errorf("noScaleUpInfo: %#v", st.PodsRemainUnschedulable[0].RejectedNodeGroups)
 				}
 				assert.Equal(t, tc.scaleUpResult, st.Result)
+
+				// Stopgap solution to ensure that fake client has time to update the status of the provisioned requests.
+				time.Sleep(1 * time.Millisecond)
+
+				provReqsAfterScaleUp, err := client.ProvisioningRequests()
+				assert.NoError(t, err)
+				assert.Equal(t, len(tc.provReqs), len(provReqsAfterScaleUp))
+
+				if tc.batchProcessing {
+					// Since batch processing returns aggregated result, we need to check the number of provisioned requests which have the provisioned condition.
+					assert.Equal(t, tc.numProvisioned, NumProvisioningRequestsWithCondition(provReqsAfterScaleUp, v1.Provisioned))
```

Comment:
Please add a check to verify if the requests are provisioned=true or provisioned=false